basis/common-router/src/main/java/com/softgroup/common/router/api/factories/TypeHandlerFactory.java

```diff
@@ -2,16 +2,34 @@
 
 import com.softgroup.common.protocol.Request;
 import com.softgroup.common.router.api.Handler;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
 /**
  * Created by Виктор on 03.03.2017.
  */
 @Component
-public class TypeHandlerFactory<T extends Handler> extends AbstractHandlerFactory<T> {
+public class TypeHandlerFactory<T extends Handler> implements HandlerFactory<T> {
```

Comment:
@Mellifaro This is almost full duplicate of CommandHandlerFactory. Why you removed abstract factory ?